libs/go/zmscli/cli.go

```diff
@@ -60,6 +63,30 @@ func (cli Zms) buildJSONOutput(res interface{}) (*string, error) {
 	return &output, nil
 }
 
+func (cli Zms) buildYAMLOutput(res interface{}) (*string, error) {
+	yamlOutput, err := yaml.Marshal(res)
+	if err != nil {
+		return nil, fmt.Errorf("failed to produce YAML output: %v", err)
+	}
+	output := string(yamlOutput)
+	return &output, nil
+}
+
+type YamlConverter func(res interface{}) (*string, error)
```

Comment:
For old (manual) YAML converter, I'll wrap the logic in a "function pointer" I can pass to dumpByFormat and just call it if the old type is used.
Once we get rid of the old yaml dump we can remove this type along with the old dump logic.